python/thirdweb-ai/src/thirdweb_ai/services/insight.py

```
@@ -409,3 +418,163 @@
         signature = validate_signature(signature)
         out = self._get(f"resolve/{signature}", params)
         return clean_resolve(out)
+
+    @tool(
+        description="Aggregate blockchain events with powerful grouping and aggregation options. Use this to get event counts, sums, or other aggregations grouped by fields like address, block, or time period."
+    )
+    def aggregate_events(
```

Pylint (3.3.7)

[refactor] Too many arguments (9/5)

(R0913)